### ISSUE TITLE
Create a latest_format file in version/

### ISSUE THREAD

**Issue by otaviobp** [clearlinux/mixer-tools#467]
There's no way to know the last available version of Clearlinux just looking at content of meta-data generated by mixer.

Currently swupd knows checks the format file in the system or uses the default format file to get the ${SERVER}/version/format${FORMAT_NUMBER}/latest and check what's the latest version to be installed. If we don't know the latest format there's no way to proceed.

With that file we could have the following command in swupd:

    $ swupd verify --install -m latest -F latest --path /my_root_dir/

That is particularly useful to create images from custom built mixers.

This issue is particularly important for swupd right now because we need this information to automate some test scripts.

**Comment by otaviobp** [comment]
What if the mixer was published in an http server and not in my system? Isn't this a use case?

And what about the official clearlinux content? How do I know the last format?

**Comment by otaviobp** [comment]
My use case: I want to install the last available release published in a server in a chroot.

Is there a way to do that without the lastest_format file?

**Comment by phmccarty** [comment]
@otaviobp I'm not thinking of a good approach to solve the problem you are outlining.

swupd-client can only understand a single format, so if it was able to consume a `${SERVER}/version/latest_format` when `latest_format` is greater than the client format, and there were some backwards incompatible changes in `latest_format`, swupd-client might not be able to consume the update content, depending on the nature of the breaking changes in the new format.

In other words, swupd-client would need to be forward compatible with any potential "breaking" changes to the update content produced by mixer.

**Comment by otaviobp** [comment]
> swupd-client can only understand a single format, so if it was able to consume a ${SERVER}/version/latest_format when latest_format is greater than the client format, and there were some backwards incompatible changes in latest_format, swupd-client might not be able to consume the update content, depending on the nature of the breaking changes in the new format.

You have a point here. Maybe we should then have in swupd codebase any information about the format swupd supports. We don't have this right now.

I'll close this issue because you convinced me.

**Comment by otaviobp** [comment]
I found a use case for this request, so I am reopening the ticket.
In swupd it would be nice to have the last_format/last_version listed in swupd check-update. This was requested in https://github.com/clearlinux/swupd-client/issues/482 and the best way to implement that is having that file.

**Comment by cmarcelo** [comment]
I support having either the latest_format or the latest_version available *somewhere* would be useful for a couple of cases for me:

- In `swupd-inspector`, we currently have to pass the version, but would be nice to inspect the latest version as convenience. For example, right now I have to say: `swupd-inspector cat clear/27120 Manifest.MoM` (note clear is a shortcut for the official Clear Linux URL, but this also works with mixes) or `swupd-inspector cat https://example.com/update/12340 Manifest.kernel-example`, but can't

- In `swupd-extract` I've ended up adding a special case for the main clear repository (so we know how to peek at "latest" file, even if its outside the update directory), but mixes need to always provide the version. This is used to create Clear container images by `mkosi`.

With latest_version I can peek at the `update/<latest_version>/format` to get the latest format.
With latest_format I can peek at the `update/version/<latest_format>/latest` to get the latest version.

For practical reasons I'd prefer to have `update/latest_version`, but either works for me.

**Comment by cmarcelo** [comment]
For the record, after suggestion from Otavio, I'd also be fine if both `update/latest_version` and `update/latest_format` (or similar) existed.

**Comment by mbelluzzo** [comment]
That help a lot with tooling automation from devops side (either mainline or downstream).

**Comment by cmarcelo** [comment]
For Clear Linux (not mixes), the release team will probably want to have more control of those values, i.e. we produce newer versions but still want users staying with the old ones until the newer are consider good to go.
